specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,1002 @@
+
+        /// Optional. Sets the version in the AppxBundleManifest.xml
+        /// If version is 0, the version is created based on the current time.
+        Windows.ApplicationModel.PackageVersion Version;
+
+        /// Overwrite output file if it already exists.
+        /// Defaults to true.
+        Boolean OverwriteOutputFileIfExists;
+    };
+
+    /// Options for unpacking a package
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass ExtractPackageOptions
+    {
+        /// Created by callers to pass options into ExtractPackage()
+        ExtractPackageOptions();
+
+        /// If true, the operation will overwrite existing files in the output path
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        Boolean OverwriteOutputFilesIfExists;
+    };
+
+    /// Options for unpacking a package
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass ExtractBundleOptions
+    {
+        /// Created by callers to pass options into ExtractBundle()
+        ExtractBundleOptions();
+        
+        /// If true, the operation will overwrite existing files in the output path
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        Boolean OverwriteOutputFilesIfExists;
+    };
+
+    /// Options for creating a mountable image
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreateMountableImageOptions
+    {
+        /// Created by callers to pass options into CreateMountableImage()
+        CreateMountableImageOptions();
+        
+        /// Root directory path for packages inside the image.
+        /// Defaults to "WindowsApps"
+        String PackageRootDirectoryInImage;
+
+        /// Overwrite output file if it already exists.
+        /// Defaults to true.
+        Boolean OverwriteOutputFileIfExists;
+    };
+
+	/// Identity of a package
+    runtimeclass PackageIdentity
+    {
+        /// Name of the package
+        String Name{ get; };
+        
+        /// Publisher of the package
+        String Publisher{ get; };
+        
+        /// Version of the package
+        Windows.ApplicationModel.PackageVersion Version{ get; };
+        
+        /// Architecture of the package
+        Windows.System.ProcessorArchitecture Architecture{ get; };
+        
+        /// ResourceId of the package
+        String ResourceId{ get; };
+
+        /// FamilyName of the package
+        String FamilyName{ get; };
+ 
+        /// FullName of the package.
+        String FullName{ get; };
+    }
+
+	/// Information about the contents of a package
+    runtimeclass PackageInformation
+    {
+		/// Identity of the package
+        PackageIdentity Identity{ get; };
+
+        /// Languages supported by the package
+        Windows.Foundation.Collections.IVectorView<String> Languages{ get; };
+
+        /// ScaleFactors supported by the package
+        Windows.Foundation.Collections.IVectorView<UInt32> ScaleFactors{ get; };
```

Comment:
AppxManifest has a 3rd `<Resource>` property, `DXFeatureLevel`. Should be added?

https://learn.microsoft.com/uwp/schemas/appxpackage/uapmanifestschema/element-resource

Reply:
Thought this was unlikely to ever be relevant for CreateKozaniPackage, but yeah, for completeness better to just have it.